src/lazy/encoder/text/v1_0/value_writer.rs

```diff
@@ -684,13 +700,28 @@ impl<'value, W: Write> ValueWriter for TextValueWriter_1_0<'value, W> {
     }
 
     fn list_writer(self) -> IonResult<Self::ListWriter> {
-        TextListWriter_1_0::new(self.writer, self.depth + 1, self.value_delimiter)
+        TextListWriter_1_0::new(
+            self.writer,
+            self.depth,
+            self.parent_type,
+            self.value_delimiter,
```

Comment:
🗺️ The container writers now know what kind of parent they appear in. This could be used to infer the right `value_delimiter`, but the `value_delimiter` logic was already there/working so I haven't removed it. Getting rid of it eliminates some code, but it also introduces a branch needed to do the inference. I don't feel strongly about it either way.